### ISSUE TITLE
CodecPool log spam

### ISSUE THREAD

**Issue by devinrsmith** [deephaven/deephaven-core#2803]
Scrolling through a parquet file, a bunch of "Got brand-new decompressor" log lines shows up:

```
2022-09-03T02:31:57.070Z | heduler-Concurrent-4 |  INFO | o.a.h.i.c.CodecPool       | Got brand-new decompressor [.zstd]
2022-09-03T02:31:57.075Z | heduler-Concurrent-4 |  INFO | o.a.h.i.c.CodecPool       | Got brand-new decompressor [.zstd]
2022-09-03T02:31:57.084Z | heduler-Concurrent-4 |  INFO | o.a.h.i.c.CodecPool       | Got brand-new decompressor [.zstd]
2022-09-03T02:31:57.089Z | heduler-Concurrent-4 |  INFO | o.a.h.i.c.CodecPool       | Got brand-new decompressor [.zstd]
2022-09-03T02:31:57.093Z | heduler-Concurrent-4 |  INFO | o.a.h.i.c.CodecPool       | Got brand-new decompressor [.zstd]
2022-09-03T02:31:57.097Z | heduler-Concurrent-4 |  INFO | o.a.h.i.c.CodecPool       | Got brand-new decompressor [.zstd]
2022-09-03T02:31:57.098Z | heduler-Concurrent-4 |  INFO | o.a.h.i.c.CodecPool       | Got brand-new decompressor [.zstd]
```

It's probably not specific to zstd. I'm not sure if there is a way to use the decompressor APIs in a different manner so these log lines don't show up. If not, we should probably add an exception for CodecPool into our logback.xml configuration.

**Comment by rcaudy** [comment]
If I recall, this is from the library, not us. I wonder if @abaranec 's codec re-use PR will make it somewhat better.

**Comment by devinrsmith** [comment]
Yep, I think Hadoop compress

**Comment by devinrsmith** [comment]
Potentially related to https://github.com/deephaven/deephaven-core/pull/2792? But that is compressors, not decompressors.

**Comment by niloc132** [comment]
This is an upstream bug, and won't be fixed by #2792, despite its work to correct similar issues. It turns out that some de/compressors opt-out of being pooled, which normally would cause this message. Unfortunately, this isn't the end of the story - the `zstd` codec doesn't actually have `Compressor`/`Decompressor` instances to even mark as `@DoNotPool` - it just returns null instead of one of these instances:
https://github.com/apache/parquet-mr/blob/6add62754b3d53e30376360f8d215da004fa8096/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/codec/ZstandardCodec.java#L68-L78
```
  @Override
  public Compressor createCompressor() {
    // ZstdOutputStream calls static Zstd compressor, so no compressor is created
    return null;
  }

  @Override
  public Decompressor createDecompressor() {
    // ZstdInputStream calls static Zstd decompressor, so no decompressor is created
    return null;
  }
```

This method is called by CodecPool.getDecompressor()/getCompressor(), which logs the `Got brand-new decompressor` message even if no brand new decompressor was created:

<insert screenshot here>

Short term, we should probably just disable this log (class+level) entirely, as there is no other logging that exists in the current version in that class. Likely it would make sense to demote that log from INFO to DEBUG, or at the very least wrap it in a null check to validate that a new de/compressor actually _was_ created.

**Comment by niloc132** [comment]
Going over other repeated log lines when running ParquetTableReadWriteTest:
 * The LZO de/compressor impl we use is explicitly marked `@DoNotPool`
 * The GZIP codec will use a compressor only if a native library can be loaded, else it will return null. On the other hand, decompressing will have an instance either way, so can be pooled. Note that we don't seem to use the native libraries for this, for some reason they aren't included in the hadoop-common jar that uses them.
